controller/controller.go

```diff
@@ -8,39 +8,60 @@ import (
 	"github.com/szuecs/kube-static-egress-controller/provider"
 )
 
+type EgressConfigSource interface {
+	ListConfigs() ([]provider.EgressConfig, error)
+	Config() <-chan provider.EgressConfig
+}
+
 // EgressController is the controller for creating Egress configuration via a
 // provider.
 type EgressController struct {
-	interval     time.Duration
-	configsChan  <-chan provider.EgressConfig
-	configsCache map[provider.Resource]map[string]struct{}
-	provider     provider.Provider
+	interval         time.Duration
+	configSource     EgressConfigSource
+	configsCache     map[provider.Resource]map[string]struct{}
+	provider         provider.Provider
+	cacheInitialized bool
 }
 
 // NewEgressController initializes a new EgressController.
-func NewEgressController(prov provider.Provider, configsChan <-chan provider.EgressConfig, interval time.Duration) *EgressController {
+func NewEgressController(prov provider.Provider, configSource EgressConfigSource, interval time.Duration) *EgressController {
 	return &EgressController{
 		interval:     interval,
-		configsChan:  configsChan,
 		provider:     prov,
+		configSource: configSource,
 		configsCache: make(map[provider.Resource]map[string]struct{}),
 	}
 }
 
 // Run runs the EgressController main loop.
 func (c *EgressController) Run(ctx context.Context) {
 	log.Info("Running controller")
-	interval := c.interval
+
 	for {
+		if !c.cacheInitialized {
+			configs, err := c.configSource.ListConfigs()
+			if err != nil {
+				log.Errorf("Failed to list Egress configurations: %v", err)
+				continue
```

Comment:
this can cause a hot loop, maybe `time.Sleep(3*time.Seconds)` or some exp backoff?